src/megatron/bridge/models/conversion/model_bridge.py

```diff
@@ -104,13 +123,43 @@ class WeightConversionTask(Generic[MappingT]):
     """
 
     param_name: str
+    global_param_name: str
     mapping: MappingT
     pp_rank: Optional[int] = None
     vp_stage: Optional[int] = None
     megatron_module: Optional[torch.nn.Module] = None
     param_weight: Optional[torch.Tensor] = None
 
 
+@dataclass(frozen=True)
+class AdapterWeightConversionTask:
+    """Task describing an adapter's LoRA weights for conversion or merging.
+
+    The task reuses :class:`WeightConversionTask` to gather the adapter's
+    linear_in/linear_out weights (if they are tensor-parallel) and carries the
+    adapter metadata required by the merge step.
+    """
+
+    global_base_prefix: str
+    adapter_key: Optional[str]  # For canonical LoRA only
+    alpha: int
+    dim: int
+    linear_in_task: WeightConversionTask
+    linear_out_task: WeightConversionTask
+
+
+@dataclass(frozen=True)
+class AdapterWeight:
```

Comment:
quite a few duplicated here and above. are they both needed. is it something close to MegatronWeightTuple(NamedTuple):?

Reply:
it's okay to duplicated, just confirm, i see you passed in only weight in merge.

Reply:
> Yeah, I guess I can turn the `linear_in_weight` and `linear_out_weight` properties of `AdapterWeight` into `MegatronWeightTuple` instead of just `torch.Tensor` for better abstraction.

Done